assignment-1/IPL.java

```diff
@@ -0,0 +1,329 @@
+    static ArrayList<Team> teams = new ArrayList<>();
+    static ArrayList<Match> matches = new ArrayList<>();
+
+    public static void fortyWicketsClub(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
+                for (Player p : team.squad) {
+                    if (p.wickets > 40) {
+                        p.display();
+                    }
+                }
+            }
+        }
+    }
+
+    public static void searchPlayersByName(String key) {
+        for (Team team : teams) {
+            for (Player p : team.squad) {
+                if (p.playerName.contains(key)) {
+                    p.display();
+                }
+            }
+
+        }
+    }
+
+    public static void getHighestWicketTakerOfTeam(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
+                Integer maxTillNow = 0;
+                Player bestBowler = null;
+                for (Player p : team.squad) {
+                    if (p.wickets >= maxTillNow) {
+                        bestBowler = p;
+                        maxTillNow = p.wickets;
+                    }
+                }
+                assert bestBowler != null;
+                bestBowler.display();
+            }
+
+        }
+    }
+
+    public static void getHighestRunScorerOfTeam(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
+                Integer maxTillNow = 0;
+                Player bestScorer = null;
+                for (Player p : team.squad) {
+                    if (p.runs >= maxTillNow) {
+                        bestScorer = p;
+                        maxTillNow = p.runs;
+                    }
+                }
+                assert bestScorer != null;
+                bestScorer.display();
+            }
+        }
+    }
+
+    public static void gettingTopPerformers() {
+        ArrayList<Player> top3Batsmen = getTopPlayersByRole("BATSMEN");
+        ArrayList<Player> top3Bowler = getTopPlayersByRole("BOWLER");
+        ArrayList<Player> top3AllRounder = getTopPlayersByRole("ALL ROUNDER");
+        displayArrayList(top3Batsmen);
+    }
+
+    static void displayArrayList(ArrayList<Player> players) {
+        for (Player player : players) {
+            player.display();
+        }
+    }
+
+    public static ArrayList<Player> getTopPlayersByRole(String role) {
+        Stack<Player> top3 = new Stack<>();
+        ArrayList<Player> result = new ArrayList<>();
+        for (Team team : teams) {
+            for (Player p : team.squad) {
+                if (p.role.equals(role)) {
+                    if (top3.size() == 0) {
+                        top3.push(p);
+                    } else {
+                        Stack<Player> temp = new Stack<>();
+                        while (!top3.isEmpty() && top3.peek().runs < p.runs) {
```

Comment:
bowlers, and all-rounders are being compared based on runs? this doesn't give the correct output